### PR TITLE
Added functionality to capture logs from Plebian and log them in Monarch

### PR DESCRIPTION
**Pull request by pawelvds**
- [ ] You've read the [Contributor Guide](https://github.com/fiskaltrust/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/fiskaltrust/.github/blob/main/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

{Summary of the changes}

## Description

{Detail}

Fixes #{issue number}

### CODE REVIEW

**Review by pawelvds — COMMENTED** [review]
In the last [commit ](https://github.com/fiskaltrust/middleware-launcher/pull/114/commits/521eafabdf4c4fb14a8683b731e842cd9ded2b61) version of the ProcessHostMonarch class, I implemented a cleaner and more modern coding style, along with improvements to logging and error handling.

Process initialisation: The process is now initialised using a more concise and readable object initialiser. The change from ```_process = new Process(); to _process = new Process { StartInfo = new ProcessStartInfo }``` is more in line with modern C# practices, making the code cleaner and easier to follow.

Buffered output and error streams: The standard output and error streams are now explicitly set to buffer. This ensures that we capture all output for analysis.
When messages are logged from the buffer, the buffer is cleared to avoid logging the same messages again if the process is restarted.

File: src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs [hunk 1]
```diff
@@ -23,50 +24,55 @@ public class ProcessHostMonarch : IProcessHostMonarch
...
-            _started = new TaskCompletionSource();
-
-            _process.OutputDataReceived += (sender, e) => 
+            _process.OutputDataReceived += (sender, e) =>
             {
-                if (e.Data != null) 
+                if (e.Data != null)
                 {
-                    _logger.LogInformation($"Plebian Output: {e.Data}");
+                    _plebianLogBuffer.Add($"Plebian Output: {e.Data}");
```
Comment (volllly) [inline src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs:62]:
```suggestion
                    _plebianLogBuffer.Add(e.Data);
```

File: src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs [hunk 2]
```diff
@@ -88,6 +94,23 @@ public Task Start(CancellationToken cancellationToken)
...
+                            _logger.LogError(log);
+                        }
+                        else
+                        {
+                            _logger.LogInformation(log);
+                        }
+                    }
+                }
+
+                _plebianLogBuffer.Clear();
```
Comment (volllly) [inline src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs:112]: We need to move this logic down to line 120 into the try catch block.

Because with the `if (!cancellationToken.IsCancellationRequested)` in line 115 we make sure this is not done when shut down the launcher with a `ctrl+c`.

File: src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs [hunk 3]
```diff
@@ -88,6 +94,23 @@ public Task Start(CancellationToken cancellationToken)
...
+                    {
+                        if (log.Contains("Error"))
+                        {
+                            _logger.LogError(log);
+                        }
+                        else
+                        {
+                            _logger.LogInformation(log);
+                        }
+                    }
```
Comment (volllly) [inline src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs:109]: I'm thinking that maybe its better to print that as one large error message instead. 🤔

something like this `_logger.LogError("Error when starting {package} {id}.\n{error}", _packageConfiguration.Package, _packageConfiguration.Id, _plebianLogBuffer.Aggregate())`


Because for one this can only happen when the plebian crashed so everything part of that error.
And we keep the information which package encountered the error inside one log message.

Ignore that comment if you disagree with my take ^^

File: src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs
Comment (volllly) [inline src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs:1]: we should probably also clear the `_plebianLogBuffer` in the `Started()` method after calling `SetResult()` to be save.

We'll need to clean up this whole class to make restart the logic more robust and also clear to understand.

Its way to hard to follow whats going on at the moment.

File: src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs (hunk 3 above)
Comment (pawelvds) [reply src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarch.cs:109]: Yes, totally agree with you. I am doing it like this. Thanks Paul

**Comment by volllly** [comment]
Very sorry to do that but I've noticed while testing that the whole restart mechanism did not work correctly (the restart actually failed and we did not actually set the error code) so I've revamped it and Fixed #116 in the process 😅

**Review by pawelvds — COMMENTED** [review]
Awesome changes, great job

**Review by volllly — APPROVED** [review]
